File: cse-308/offline-1/problem-2/Car.java
public abstract class Car
{
    protected String brand;
    protected Color color;
    protected Engine engine;
    protected WheelType wheelType;

    public String GetBrand()
    {
        return brand;
    }

    public Color GetColor()
    {
        return color;
    }

    public Engine GetEngine()
    {
        return engine;
    }

    public WheelType GetWheelType()
    {
        return wheelType;
    }
}

File: cse-308/offline-1/problem-2/src/Car.cpp
#include "Car.h"

Car::Car()
{
    name = "";
    color = NULL;
    engine = NULL;
    driveTrain = NULL;
}

std::string Car::GetName()
{
    return name;
}

Color *Car::GetColor()
{
    return color;
}

Engine *Car::GetEngine()
{
    return engine;
}

DriveTrain *Car::GetDriveTrain()
{
    return driveTrain;
}

Car::~Car()
{
    delete color;
    delete engine;
    delete driveTrain;
}